crates/ra_ide_api/src/expand_macro.rs

```diff
@@ -269,4 +269,27 @@ fn some_thing() -> u32 {
         assert_eq!(res.name, "foo");
         assert_snapshot!(res.expansion, @r###"bar!()"###);
     }
+
+    #[test]
+    fn macro_expand_with_dollar_crate() {
+        let res = check_expand_macro(
+            r#"
+        //- /lib.rs
+        #[macro_export]
+        macro_rules! bar {
+            () => {0};
+        }
+        macro_rules! foo {
+            () => {$crate::bar!()};
+        }        
+
+        fn main() {        
+            let res = fo<|>o!();
+        }
+        "#,
+        );
+
+        assert_eq!(res.name, "foo");
+        assert_snapshot!(res.expansion, @r###"0"###);
```

Comment:
Could you also test invoking `foo` from a crate other than where it is defined?

Reply:
I don't think this is really necessary: there should be other test that checks that `$crate` is handled correctly, if the hygiene is present. But this wouldn't hurt either!